src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -23,6 +23,7 @@
 import java.util.Objects;
 import java.util.function.Supplier;
 
+import lombok.Builder;
```

Comment:
I think we want to validate the NeuralQueryBuilder before we create it like what we are doing in the KNNQueryBuilder. And to enforce this we will also need to either remove @NoArgsConstructor
@AllArgsConstructor or make them private.

Currently I think the builder is mainly used for our internal testing but I think it's good to enforce that only valid NeuralQueryBuilder can be created through our builder in case in future people accidentally use it create some invalid NeuralQueryBuilder in our business logic.

Reply:
Besides I think currently we are using AllArgsConstructor to create NeuralQueryBuilder for some tests can we replace them with the builder you add? In this way if we need to add a new field we don't need to modify the tests that don't care about the new field.

Reply:
+1 to @bzhangam's comment. You can have reference from kNN's query builder: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java#L142